build/dsc/templates/dsc_type.rb.erb

```diff
@@ -103,6 +103,10 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
     defaultto { :<%= begin resource.ensure_value rescue 'present' end %> }
   end
 
+if Puppet.version >= '4.0.0'
```

Comment:
So the problem with this is that we also want an edge to exist prior to Puppet 4.

Thinking about this a little more, it may be easier for us (given Puppet 3.x is the mix) to just write a little code similar to what's in here https://github.com/puppetlabs/puppet/blob/d2199af906fa5ca19ec0c37c82794934ea4fb4b5/lib/puppet/type.rb#L2111-L2143

The important part is mostly finding the resource and doing `reqs << Puppet::Relationship.new(self, dep)`

Then we can do the edge manually instead of using `autonotify` syntax.